### ISSUE TITLE
Explorer / Lisk Service backend code refactoring

### ISSUE THREAD

**Issue by MichalTuleja** [LiskHQ/lisk-explorer#461]
In order to make the Explorer backend code an independent part called Lisk Service a lot of refactoring tasks should be done before. Achieving good run-time stability and making the codebase easy to maintain is crucial to the project existence, assuming that 3rd party tools can also use the explorer API in the future.

Most important:
 - establishing a transport layer for multiple HTTP sources - most of the requests points to our Lisk Core nodes, but there are also miscellaneous external services like crypto exchanges (currently two of them), FreeGeoIP, FIAT currency exchange rates, etc.,
 - Redis caching is sometimes unnecessary and cause a lot of problems - it may be replaced with some other solution, or kept only for limited amount of GET requests,
 - Market Watcher candles data are stored in Redis - due to volatile characteristics of the API endpoints and very low availability, a lot of things can be improved here - ie. Bittrex provides trading data for only a few hours behind, Binance only last 500 trades. It may be worth storing the data in some persistent NoSQL database and even keep gathering such data in a separate process (or application).
 - Almost no error handling for bad HTTP responses,
 - No proper data validation - random error may shut the whole process down because of an unhandled exception,
 - Although there is very little complexity (most of the job done by the backend is basically proxying), having unit tests would be also nice to have.
